homeassistant/components/fan/insteon_local.py

```diff
@@ -32,98 +28,14 @@
-
-
-def request_configuration(device_id, insteonhub, model, hass,
-                          add_devices_callback):
-    """Request configuration steps from the user."""
-    configurator = hass.components.configurator
-
-    # We got an error if this method is called while we are configuring
-    if device_id in _CONFIGURING:
-        configurator.notify_errors(
-            _CONFIGURING[device_id], 'Failed to register, please try again.')
-
-        return
-
-    def insteon_fan_config_callback(data):
-        """The actions to do when our configuration callback is called."""
-        setup_fan(device_id, data.get('name'), insteonhub, hass,
-                  add_devices_callback)
-
-    _CONFIGURING[device_id] = configurator.request_config(
-        'Insteon  ' + model + ' addr: ' + device_id,
-        insteon_fan_config_callback,
-        description=('Enter a name for ' + model + ' Fan addr: ' + device_id),
-        entity_picture='/static/images/config_insteon.png',
-        submit_caption='Confirm',
-        fields=[{'id': 'name', 'name': 'Name', 'type': ''}]
-    )
-
-
-def setup_fan(device_id, name, insteonhub, hass, add_devices_callback):
-    """Set up the fan."""
-    if device_id in _CONFIGURING:
-        request_id = _CONFIGURING.pop(device_id)
-        configurator = hass.components.configurator
-        configurator.request_done(request_id)
-        _LOGGER.info("Device configuration done!")
-
-    conf_fans = config_from_file(hass.config.path(INSTEON_LOCAL_FANS_CONF))
-    if device_id not in conf_fans:
-        conf_fans[device_id] = name
-
-    if not config_from_file(
-            hass.config.path(INSTEON_LOCAL_FANS_CONF),
-            conf_fans):
-        _LOGGER.error("Failed to save configuration file")
-
-    device = insteonhub.fan(device_id)
-    add_devices_callback([InsteonLocalFanDevice(device, name)])
-
-
-def config_from_file(filename, config=None):
-    """Small configuration file management function."""
-    if config:
-        # We're writing configuration
-        try:
-            with open(filename, 'w') as fdesc:
-                fdesc.write(json.dumps(config))
-        except IOError as error:
-            _LOGGER.error('Saving config file failed: %s', error)
-            return False
-        return True
-    else:
-        # We're reading config
-        if os.path.isfile(filename):
-            try:
-                with open(filename, 'r') as fdesc:
-                    return json.loads(fdesc.read())
-            except IOError as error:
-                _LOGGER.error("Reading configuration file failed: %s", error)
-                # This won't work yet
-                return False
-        else:
-            return {}
+    linked = discovery_info['linked']
+
+    for device_id in linked:
+        if (linked[device_id]['cat_type'] == 'dimmer' and
+                linked[device_id]['sku'] == '2475F'):
+            _LOGGER.info("Adding fan device " + device_id)
```

Comment:
I think we could probably drop this log message as well, since hass core will output a log for the new entity state, which includes the device_id since we're using that as the name.